components/builder-worker/src/runner/docker.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::PathBuf;
+use std::process::{Child, Command, ExitStatus, Stdio};
+
+use hab_core::env;
+use hab_core::fs as hfs;
+use hab_core::os::process::{self, Signal};
+
+use error::Result;
+use runner::log_pipe::LogPipe;
+use runner::{NONINTERACTIVE_ENVVAR, RUNNER_DEBUG_ENVVAR};
+use runner::workspace::Workspace;
+
+lazy_static! {
+    /// Absolute path to the Docker exporter program
+    static ref DOCKER_EXPORTER_PROGRAM: PathBuf = hfs::resolve_cmd_in_pkg(
+        "hab-pkg-export-docker",
+        include_str!(concat!(env!("OUT_DIR"), "/DOCKER_EXPORTER_PKG_IDENT")),
+    );
+
+    /// Absolute path to the Dockerd program
+    static ref DOCKERD_PROGRAM: PathBuf = hfs::resolve_cmd_in_pkg(
+        "dockerd",
+        include_str!(concat!(env!("OUT_DIR"), "/DOCKER_PKG_IDENT")),
+    );
+}
+
+const DOCKER_HOST_ENVVAR: &'static str = "DOCKER_HOST";
+
+pub struct DockerExporter<'a> {
+    workspace: &'a Workspace,
+    bldr_url: &'a str,
+}
+
+impl<'a> DockerExporter<'a> {
+    /// Creates a new Docker exporter for a given `Workspace` and Builder URL.
+    pub fn new(workspace: &'a Workspace, bldr_url: &'a str) -> Self {
+        DockerExporter {
+            workspace,
+            bldr_url,
+        }
+    }
+
+    /// Spawns a Docker export command, pipes output streams to the given `LogPipe` and returns the
+    /// process' `ExitStatus`.
+    ///
+    /// # Errors
+    ///
+    /// * If the child process can't be spawned
+    /// * If the calling thread can't wait on the child process
+    /// * If the `LogPipe` fails to pipe output
+    pub fn export(&self, log_pipe: &mut LogPipe) -> Result<ExitStatus> {
+        let dockerd = self.spawn_dockerd()?;
+        let exit_status = self.run_export(log_pipe);
+        self.teardown_dockerd(dockerd)?;
```

Comment:
I would warn that this failed but I wouldn't crash the worker because of it. Ideally we pick this up in our log dashboard and the investigate why our disk is filling up but jobs are still completing :)

Reply:
Whoops, that was what I intended before I shunted it all to a method. Will update.